src/lib/support/ThreadOperationalDataset.cpp

```diff
@@ -0,0 +1,629 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <assert.h>
+#include <string.h>
+
+#include "ThreadOperationalDataset.h"
+
+namespace chip {
+namespace Thread {
+
+/**
+ * This class implements Thread Tlv functionality.
```

Comment:
Is thread TLV different than regular TLV?

Could we use code/CHIPTlv.h and similar?

Reply:
I was told that Thread has its own storage format unrelated to CHIP tag-length-value.

I propose to name this differently in this case to alleviate confusion. ThreadValue may be the simplest however maybe `TypeSizePrefixValue` is clearer (or something similar ... this name is somewhat of a mouth full)

Reply:
I believe within chip codebase using the short-to-type version leads to confusion - it totally confused me.

I prefer 'TypeLengthValue' fully spelled out instead of TLV.

Reply:
I renamed to ThreadTlv. `TypeLengthValue` confuses me.